WooCommerce/WooCommerceUITests/Screens/MyStore/MyStoreScreen.swift

```diff
@@ -1,39 +1,39 @@
-import Foundation
+import ScreenObject
 import XCTest
 
-final class MyStoreScreen: BaseScreen {
-
-    struct ElementStringIDs {
-        static let topBannerCloseButton = "top-banner-view-dismiss-button"
-        static let settingsButton = "dashboard-settings-button"
-    }
+final class MyStoreScreen: ScreenObject {
 
     let tabBar = TabNavComponent()
     let periodStatsTable = PeriodStatsTable()
-    private let settingsButton: XCUIElement
-    private let topBannerCloseButton: XCUIElement
+
+    private var settingsButton: XCUIElement { Self.settingsButtonGetter(app) }
+    private var topBannerCloseButton: XCUIElement { Self.topBannerCloseButtonGetter(app) }
 
     static var isVisible: Bool {
-        let settingsButton = XCUIApplication().buttons[ElementStringIDs.settingsButton]
-        return settingsButton.exists && settingsButton.isHittable
+        guard let screen = try? MyStoreScreen() else { return false }
+        return screen.settingsButton.isHittable
     }
 
-    init() {
-        settingsButton = XCUIApplication().buttons[ElementStringIDs.settingsButton]
-        topBannerCloseButton = XCUIApplication().buttons[ElementStringIDs.topBannerCloseButton]
+    private static let settingsButtonGetter: (XCUIApplication) -> XCUIElement = {
+        $0.buttons["dashboard-settings-button"]
+    }
 
-        super.init(element: settingsButton)
+    private static let topBannerCloseButtonGetter: (XCUIApplication) -> XCUIElement = {
+        $0.buttons["top-banner-view-dismiss-button"]
+    }
 
-        XCTAssert(settingsButton.waitForExistence(timeout: 3))
+    init(app: XCUIApplication = XCUIApplication()) throws {
+        try super.init(
+            expectedElementGetters: [Self.settingsButtonGetter, Self.topBannerCloseButtonGetter],
+            app: app
+        )
     }
 
     @discardableResult
     func dismissTopBannerIfNeeded() -> MyStoreScreen {
+        guard topBannerCloseButton.waitForExistence(timeout: 3) else { return self }
 
-        if topBannerCloseButton.waitForExistence(timeout: 3) {
-            topBannerCloseButton.tap()
-        }
-
+        topBannerCloseButton.tap()
```

Comment:
This method is the same, I simply replaced the `if` with a `guard`.

Now that I'm writing this comment, I realize the change is a bit arbitrary. I guess it just felt more appropriate to use a `guard` in combination with a "wait for existence" method than an `if`. Happy to revert if this doesn't make sense to you.